lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -59,24 +59,79 @@ def set_up_course(self, enable_subsection_grades=True):
+            enqueue_op,
             return_value=None
         ) as mock_task_apply:
-            SCORE_CHANGED.send(sender=None, **self.score_changed_kwargs)
-            mock_task_apply.assert_called_once_with(args=tuple(self.score_changed_kwargs.values()))
+            test_signal.send(sender=None, **self.score_changed_kwargs)
+            mock_task_apply.assert_called_once_with(args=tuple(expected_args))
 
-    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
-    def test_subsection_grade_updated(self, default_store):
+    @patch('lms.djangoapps.grades.signals.signals.COURSE_GRADE_UPDATE_REQUESTED.send')
+    def test_subsection_update_triggers_course_update(self, mock_course_signal):
+        """
+        Ensures that the subsection update operation also updates the course grade.
+        """
+        self.set_up_course()
+        recalculate_subsection_grade.apply(args=tuple(self.score_changed_kwargs.values()))
+        mock_course_signal.assert_called_once_with(
+            sender=recalculate_subsection_grade,
+            course_id=self.score_changed_kwargs['course_id'],
+            user_id=self.score_changed_kwargs['user_id'],
+        )
+
+    @ddt.data(True, False)
+    def test_course_update_enqueuing(self, should_be_async):
+        """
+        Ensures that the course update operation is enqueued on an async queue (or not) as expected.
+        """
+        base = 'lms.djangoapps.grades.tasks.recalculate_course_grade.apply'
+        if should_be_async:
+            execute = base + '_async'
+            other = base
+        else:
+            execute = base
+            other = base + '_async'
+        self.set_up_course()
+
+        with patch(execute, return_value=None) as task_to_execute:
+            with patch(other, return_value=None) as other_task:
+                COURSE_GRADE_UPDATE_REQUESTED.send(
+                    sender=None if should_be_async else recalculate_subsection_grade,
+                    course_id=self.score_changed_kwargs['course_id'],
+                    user_id=self.score_changed_kwargs['user_id'],
+                )
+                other_task.assert_not_called()
+                task_to_execute.assert_called_once_with(
+                    args=(
+                        self.score_changed_kwargs['user_id'],
+                        self.score_changed_kwargs['course_id'],
+                    )
+                )
+
+    @ddt.data(
+        (ModuleStoreEnum.Type.mongo, 1),
+        (ModuleStoreEnum.Type.split, 0),
+    )
+    @ddt.unpack
+    def test_subsection_grade_updated(self, default_store, added_queries):
         with self.store.default_store(default_store):
             self.set_up_course()
             self.assertTrue(PersistentGradesEnabledFlag.feature_enabled(self.course.id))
-            with check_mongo_calls(2) and self.assertNumQueries(13):
+            with check_mongo_calls(2) and self.assertNumQueries(21 + added_queries):
```

Comment:
That's a bunch of extra queries.  Are we grabbing subsections multiple times?  Can we grab them once, or even in bulk?